calendar/index.js

```diff
@@ -0,0 +1,273 @@
+
+        this.state = {
+            checkin: null,
+            checkout: null,
+        };
+
+        this.dayButtons = [];
+    }
+
+    async connectedCallback() {
+        const renderedCss = await loadAndInjectCSS('../calendar/css/calendar.css');
+        this.render(renderedCss);
+        this.loadFromSessionStorage();  // Load from sessionStorage and apply necessary classes
+
+        // Listen for custom events tied to the specific "name" attribute
+        document.addEventListener(`checkin-selected-${this.name}`, (e) => this.syncState(e));
+        document.addEventListener(`checkout-selected-${this.name}`, (e) => this.syncState(e));
+    }
+
+    disconnectedCallback() {
+        document.removeEventListener(`checkin-selected-${this.name}`, this.syncState);
+        document.removeEventListener(`checkout-selected-${this.name}`, this.syncState);
+    }
+
+    render(css) {
+        this.innerHTML = `<style>${css}</style>`;
+        this.renderInputGroup();
+        this.renderCalendarPanel();
+        this.renderHiddenInputs();
+        this.loadMonthButtons();
+    }
+
+    renderInputGroup() {
+        const inputGroup = this.createElementWithClasses('div', [CLASS_NAMES.inputGroup]);
+        const labelElement = this.createElementWithText('label', this.label);
+        const input = this.createElementWithAttributes('input', { type: 'text', placeholder: this.placeholder, readOnly: true });
+
+        this.inputElement = input;  // Store input for future use
+        inputGroup.append(labelElement, input);
+        inputGroup.addEventListener('click', () => this.toggleCalendar());
+        this.appendChild(inputGroup);
+    }
+    
+    renderCalendarPanel() {
+        this.panel = this.createElementWithClasses('div', [CLASS_NAMES.panel]);
+        this.panel.style.display = 'none';
+        this.appendChild(this.panel);
+    }
+    
+
+    renderHiddenInputs() {
+        this.startInput = this.createElementWithAttributes('input', { type: 'hidden', name: this.startName });
+        this.endInput = this.createElementWithAttributes('input', { type: 'hidden', name: this.endName });
+        this.append(this.startInput, this.endInput);
+    }
+
+    loadMonthButtons() {
+        const months = getNext12Months(this.openDate);
+        months.forEach(month => {
+            const monthEl = createMonthElement(month, this.weekDays, this.dateLocale);
+            this.panel.appendChild(monthEl);
+
+            month.days.forEach(day => {
+                const dayBtn = createDayButton(day, this.dateLocale);
+                if (day.date < this.openDate) dayBtn.disabled = true;
+
+                dayBtn.addEventListener('click', (e) => this.handleDayClick(e, dayBtn));
+                monthEl.querySelector(`.${CLASS_NAMES.daysContainer}`).appendChild(dayBtn);
+                this.dayButtons.push(dayBtn);
+            });
+        });
+    }
+
+    handleDayClick(event, dayBtn) {
```

Comment:
@samuelsilvapt I wonder if breaking this out into smaller functions might be easier to read?